makefiles/suit.v4.inc.mk

```diff
@@ -20,33 +20,40 @@ SUIT_DEVICE_ID ?= $(BOARD)
 #
 # SUIT encryption keys
 #
+
+# Specify key to use.
+# Will use $(SUIT_KEY_DIR)/$(SUIT_KEY) $(SUIT_KEY_DIR)/$(SUIT_KEY).pub as
+# private/public key files, similar to how ssh names its key files.
+SUIT_KEY ?= default
+
 ifeq (1, $(RIOT_CI_BUILD))
   SUIT_KEY_DIR ?= $(BINDIR)
 else
   SUIT_KEY_DIR ?= $(RIOTBASE)/keys
 endif
 
-SUIT_KEY ?= $(SUIT_KEY_DIR)/secret.key
-SUIT_PUB ?= $(SUIT_KEY_DIR)/public.key
+SUIT_SEC ?= $(SUIT_KEY_DIR)/$(SUIT_KEY)
```

Comment:
Maybe replace `?=` by `=` or `:=` (I'm not sure the former will work). Same for `SUIT_PUB`

Reply:
why? no need to be restrictive here.

Reply:
Because we don't want to expose these files names to the user since it may end up in inconsistent/broken private/public key pair.
The introduction of `SUIT_KEY` is here to ensure the public and private key pair is valid.

Reply:
There are use-cases where someone might be in possession of the public key file and wants to build an image, but does not possess the private key. The machinery for the public key header only needs the public key file. Thus, IMO, it makes sense to be able to just plainly specify the public key file on the build command line.



> The introduction of `SUIT_KEY` is here to ensure the public and private key pair is valid.

Rather to support a well-known naming scheme and not have the user to specify two full paths, in the default case.

Common, having ```?=``` or ```=``` is not worth discussing at this point.

Reply:
> There are use-cases where someone might be in possession of the public key file and wants to build an image, but does not possess the private key.

Can you detail some of them?
In the SUIT context, I don't see such use case since both keys are required: the public key for building the image and the private key for signing the manifest associated to this image.